lib/findOne.js

```diff
@@ -1,35 +1,25 @@
 const _ = require('lodash');
 
 var defaultSerialise = require('./serialise');
+var parseInclude = require('./parseInclude');
 
 module.exports = function(options) {
   return function(req, res, next) {
-    return new Promise(function(resolve) {
-      if (options.translateId) {
-        return resolve(options.translateId(req.params.id, req));
-      }
+    var id = options.translateId ? options.translateId(req.params.id, req) : req.params.id;
+    var relInclude = parseInclude(req, options);
+    var queryObject = req.autorouteQuery || {};
 
-      return resolve(req.params.id);
-    })
+    if (options.idParameter || options.find.idParameter) {
+      queryObject[options.idParameter || options.find.idParameter] = id;
+    } else {
+      queryObject._id = new options.model.base.Types.ObjectId(id);
+    }
 
-      .then(function(id) {
-        var queryObject = req.autorouteQuery || {};
-
-        if (options.idParameter || options.find.idParameter) {
-          queryObject[options.idParameter || options.find.idParameter] = id;
-        } else {
-          queryObject._id = new options.model.base.Types.ObjectId(id);
-        }
-
-        var query = options.model.findOne(queryObject);
-
-        if (options.find.populate) {
-          query.populate(options.find.populate);
```

Comment:
This line wasn't working without further changes to the serialiser. I removed it. Look how clean the code is! It's double the functionality in half the lines! It's beautiful! :cry: